OpenRA.Game/Network/OrderIO.cs

```diff
@@ -109,37 +133,38 @@ public static bool TryParseSync(byte[] packet, out (int Frame, int SyncHash, ulo
 			return true;
 		}
 
-		public static bool TryParsePing(int fromClient, byte[] packet, out byte[] ping)
+		public static bool TryParsePingRequest(int fromClient, byte[] packet, out long timestamp)
 		{
-			// Valid Ping packets are only ever generated by the server
+			// Valid Ping requests are only ever generated by the server
 			if (fromClient != 0 || packet.Length != 13 || packet[4] != (byte)OrderType.Ping)
 			{
-				ping = null;
+				timestamp = 0;
 				return false;
 			}
 
 			// Valid Ping packets always have frame 0
 			var frame = BitConverter.ToInt32(packet, 0);
 			if (frame != 0)
 			{
-				ping = null;
+				timestamp = 0;
 				return false;
 			}
 
-			ping = packet;
+			timestamp = BitConverter.ToInt64(packet, 5);
 			return true;
 		}
 
-		public static bool TryParseAck((int FromClient, byte[] Data) packet, out int frame)
+		public static bool TryParseAck((int FromClient, byte[] Data) packet, out int frame, out byte count)
 		{
 			// Ack packets are only accepted from the server
-			if (packet.FromClient != 0 || packet.Data.Length != 5 || packet.Data[4] != (byte)OrderType.Ack)
+			if (packet.FromClient != 0 || packet.Data.Length != 6 || packet.Data[4] != (byte)OrderType.Ack)
```

Comment:
We will never be acknowledging more than a couple of frames, so I felt that the simplest tradeoff was to always send a byte instead of optionally sending an int.

Reply:
What happens if a game is paused? No orders, so no acks?

Reply:
Frames continue to advance while the game is paused. They only stop if traffic is stalled.